src/microbit/test/test_display.py

```diff
@@ -0,0 +1,148 @@
+            self.display.set_pixel(x, y, brightness)
+
+    def test_clear(self):
+        self.display._Display__image._Image__LED[2][3] = 7
+        self.display._Display__image._Image__LED[3][4] = 6
+        self.display._Display__image._Image__LED[4][4] = 9
+        assert not self.__is_clear()
+        self.display.clear()
+        assert self.__is_clear()
+
+    def test_on(self):
+        self.display._Display__on = False
+        self.display.on()
+        assert self.display._Display__on
+
+    def test_off(self):
+        self.display._Display__on = True
+        self.display.off()
+        assert False == self.display._Display__on
+
+    @pytest.mark.parametrize("on", [True, False])
+    def test_is_on(self, on):
+        self.display._Display__on = on
+        assert on == self.display.is_on()
+
+    def test_show_one_image(self):
+        img = Image()
+        img.set_pixel(0, 0, 8)
+        img.set_pixel(0, 1, 9)
+        img.set_pixel(0, 2, 7)
+        img.set_pixel(2, 2, 6)
+        self.display.show(img)
+        assert Image._Image__same_image(img, self.display._Display__image)
+
+    def test_show_different_size_image(self):
+        img = Image(3, 7)
+        img.set_pixel(1, 1, 9)
+        img.set_pixel(2, 6, 9)  # Will not be on display
+        expected = Image(5, 5)
+        expected.set_pixel(1, 1, 9)
+        self.display.show(img)
+        assert Image._Image__same_image(expected, self.display._Display__image)
+
+    def test_show_smaller_image(self):
+        img = Image(2, 2)
+        img.set_pixel(1, 1, 9)
+        expected = Image(5, 5)
+        expected.set_pixel(1, 1, 9)
+        self.display.show(img)
+        assert Image._Image__same_image(expected, self.display._Display__image)
+
+    @pytest.mark.parametrize(
+        "value, expected_str",
+        [
+            ("!", STR_EXCLAMATION_MARK),
+            ("A", STR_A),
+            (" ", CONSTANTS.BLANK_5X5),
+            (6, STR_SIX),
+            ("\x7F", STR_QUESTION_MARK),  # Character is out of our ASCII range
+        ],
+    )
+    def test_show_char(self, value, expected_str):
+        expected = Image(expected_str)
+        self.display.show(value)
+        assert Image._Image__same_image(expected, self.display._Display__image)
+
+    def test_show_char_with_clear(self):
+        image = Image(STR_EXCLAMATION_MARK)
+        self.display.show(image, clear=True)
+        assert self.__is_clear()
+
+    def test_show_iterable(self):
+        expected = Image(STR_A)
+        value = [Image(STR_EXCLAMATION_MARK), "A", "ab"]
+        self.display.show(value)
+        assert Image._Image__same_image(expected, self.display._Display__image)
+
+    def test_show_non_iterable(self):
+        with pytest.raises(TypeError):
+            self.display.show(TestDisplay())
+
+    def test_show_threaded(self):
+        threading.Thread = mock.Mock()
+        self.display.show("a", wait=False)
+        threading.Thread.assert_called_once()

```

Comment:
I would remove this test (or even better change it so it tests behavior), since it tests the implementation.
For ex: if we switch to `async` in the future, this would fail even though maybe everything is working correctly.
Same for `test_scroll_threaded`.

Reply:
I'm going to leave it for now but it is definitely something to look back to in the future.